core/arch/arm/tee/svc_cache.c

```diff
@@ -60,5 +59,5 @@ TEE_Result syscall_cache_operation(void *va, size_t len,
 	if (res != TEE_SUCCESS)
 		return TEE_ERROR_ACCESS_DENIED;
 
-	return cache_operation(op, va, len);
+	return cache_operation((enum utee_cache_operation)op, va, len);
```

Comment:
Unnecessary casting could hide some warning and also clutters the code in my opinion. I doubt you'll find much of this pattern in the Linux kernel.